app/src/main/java/ch/hikemate/app/ui/auth/CreateAccountScreen.kt

```diff
@@ -0,0 +1,150 @@
+                  top = 16.dp,
+              ),
+      verticalArrangement = Arrangement.spacedBy(16.dp)) {
+        BackButton(navigationActions)
+        Text(
+            stringResource(R.string.create_account_title),
+            style = TextStyle(fontWeight = FontWeight.Bold, fontSize = 32.sp),
+            modifier = Modifier.testTag(CreateAccountScreen.TEST_TAG_TITLE))
+
+        OutlinedTextField(
+            modifier = Modifier.fillMaxWidth().testTag(CreateAccountScreen.TEST_TAG_NAME_INPUT),
+            colors = inputColors,
+            value = name,
+            onValueChange = { name = it },
+            label = { Text(stringResource(R.string.create_account_name_label)) })
+
+        OutlinedTextField(
+            modifier = Modifier.fillMaxWidth().testTag(CreateAccountScreen.TEST_TAG_EMAIL_INPUT),
+            keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Email),
+            colors = inputColors,
+            value = email,
+            onValueChange = { email = it },
+            label = { Text(stringResource(R.string.create_account_email_label)) })
+
+        OutlinedTextField(
+            modifier = Modifier.fillMaxWidth().testTag(CreateAccountScreen.TEST_TAG_PASSWORD_INPUT),
+            visualTransformation = PasswordVisualTransformation(),
+            colors = inputColors,
+            value = password,
+            onValueChange = { password = it },
+            label = { Text(stringResource(R.string.create_account_password_label)) })
+
+        OutlinedTextField(
+            modifier =
+                Modifier.fillMaxWidth()
+                    .testTag(CreateAccountScreen.TEST_TAG_CONFIRM_PASSWORD_INPUT),
+            visualTransformation = PasswordVisualTransformation(),
+            colors = inputColors,
+            value = confirmPassword,
+            onValueChange = { confirmPassword = it },
+            label = { Text(stringResource(R.string.create_account_repeat_password_label)) })
+
+        BigButton(
+            modifier = Modifier.fillMaxWidth().testTag(CreateAccountScreen.TEST_TAG_SIGN_UP_BUTTON),
+            buttonType = ButtonType.PRIMARY,
+            label = stringResource(R.string.create_account_create_account_button),
+            onClick = {
+              if (password != confirmPassword) {
+                // Show an error message in a toast
+                Toast.makeText(context, mismatchErrorMessage, Toast.LENGTH_SHORT).show()
+              } else {
+                authViewModel.createAccountWithEmailAndPassword(
+                    email,
+                    password,
+                    onSuccess = {
+                      // Navigate to the map screen
+                      navigationActions.navigateTo(Route.MAP)
+                    },
+                    onErrorAction = {
+                      // Show an error message in a toast
+                      Toast.makeText(context, it.message, Toast.LENGTH_SHORT).show()
+                    })
+              }
```

Comment:
Nitpick: For cleanliness, maybe just define an inner function called `onButtonPressed` that could be defined still in the CreateAccountScreen function, but higher up than in the Composable architecture, and call it here.

Reply:
Nice idea. Thanks!